ir/ir.rs

```diff
@@ -378,6 +378,11 @@ impl Id {
     pub fn root() -> Id {
         Id(u32::MAX, u32::MAX)
     }
+
+    /// Get the real id of a item.
+    pub fn real_id(&self) -> u32 {
```

Comment:
It is not clear to me what this is for. Both components are real, and dropping one or the other loses the unique-ness property.

Reply:
Since I don't know why the `Id` is represented by a u32-tuple(my fault, I should ask for your advice before coding). So how can I write a `Id`? Just raw tuple as it's or?

Reply:
It is a pair of `(section index, item within that section index)`

To get something that could be serialized as a single thing, you could combine them into a `u64`:

```rust
let top = self.0 as u64 << 32;
top | (self.1 as u64)
```

I would call this `serializable` or something rather than `real_id`.